quantum/pointing_device/pointing_device_auto_mouse.h

```diff
@@ -0,0 +1,91 @@
+/* Copyright 2022 Alabastard
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <string.h>
+
+#include "quantum.h"
+#include "pointing_device.h"
+#include "print.h"
+
+/* check settings and set defaults */
+#ifndef POINTING_DEVICE_AUTO_MOUSE_ENABLE
+#    error "POINTING_DEVICE_AUTO_MOUSE_ENABLE not defined! check config settings"
+#endif
+
+#ifndef AUTO_MOUSE_DEFAULT_LAYER
+#    define AUTO_MOUSE_DEFAULT_LAYER 1
+#endif
+#ifndef AUTO_MOUSE_TIME
+#    define AUTO_MOUSE_TIME 650
+#endif
+#ifndef AUTO_MOUSE_DELAY
+#    ifdef TAPPING_TERM
+#        define AUTO_MOUSE_DELAY TAPPING_TERM
+#    else
+#        define AUTO_MOUSE_DELAY 200
+#    endif
```

Comment:
```suggestion
    #define AUTO_MOUSE_DELAY GET_TAPPING_TERM(KC_XXX, &(keyrecord_t){})
```

`TAPPING_TERM` is always defined but can be dynamic, and there is the handy `GET_TAPPING_TERM` macro now. I'm unsure which keycode to use for the dynamic tapping term though.

Reply:
IIRC, I just opted to check `KC_BTN1`

Reply:
perhaps this should be:
```c
#    GET_TAPPING_TERM(KC_NO, &(keyrecord_t){})
```
hopefully this is a surefire way to force the default

Reply:
Well, my reasoning for KC_BTN1 here is that it's a mouse related button, so is a bit more consistent with the functionality (mouse related).  I'm not too attached to the specific, but I don't think that `KC_NO` is a great choice here.

But as long as it's documented properly, then it shouldn't matter too much.

Reply:
Ah yeah makes sense just forcing a hard coded default would make more sense than using  `KC_NO`, so `KC_BTN1` is more logical and communicates intent better even if the result is the same. Will change right away.

Reply:
`KC_BTN1` is a good choice IMHO.